Front/category.cpp

```diff
@@ -0,0 +1,105 @@
+#include "category.h"
+#include "ui_category.h"
+#include "addcategory.h"
+#include "popupmenu.h"
+#include "dynamiccategory.h"
+#include "deletednotes.h"
+#include "userpage.h"
+#include "addsavenote.h"
+#include "notesincategory.h"
+
+Category::Category(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::Category)
+{
+    ui->setupUi(this);
+}
+
+Category::~Category()
+{
+    delete ui;
+}
+
+
+void Category::recieveData(QString str)
+{
+    ui->CScrollArea->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOn);
+    QGridLayout *lay = new QGridLayout(ui->CScrollArea);
+    lay->addWidget(ui->CScrollArea, 0, 0, 1, 1);
+
+    auto  gridLayout = new QGridLayout(ui->scrollAreaWidgetContents);
+    auto  widget = new QWidget(ui->scrollAreaWidgetContents);
+    widget->setObjectName(QString::fromUtf8("widget"));
+    gridLayout->addWidget(widget, 0, 0, 1, 3);
+
+
+
+    DynamicCategory *category = new DynamicCategory(this);
+    category->setText(QString(str));
+    PopupMenu* menu = new PopupMenu(category, this);
+    menu ->addAction("Open");
+    menu ->addAction("Edit");
+    menu ->addAction("Delete");
+    category->setMenu(menu);
+    category->setStyleSheet(       "color: rgb(27, 43, 66);"
+                                   "border: 2px solid  rgb(27, 43, 66);"
+                                   "border-radius: 10px;"
+                                   "padding: 1px 18px 1px 3px;"
+                                   "min-width: 6em;}"
```

Comment:
You can possibly have CSS syntax error here

Reply:
Fixed